stl/inc/xutility

```diff
@@ -3063,8 +3063,477 @@ namespace ranges {
+        template <class _Dx = _Derived>
+        _NODISCARD constexpr decltype(auto) back() const
+            requires bidirectional_range<const _Dx> && common_range<const _Dx>
+#else // ^^^ workaround / no workaround vvv
+        _NODISCARD constexpr decltype(auto) back() const
+            requires bidirectional_range<const _Derived> && common_range<const _Derived>
+#endif // TRANSITION, LLVM-44833
+        {
+            auto& _Self = _Cast();
+#if _CONTAINER_DEBUG_LEVEL > 0
+            _STL_VERIFY(!_RANGES empty(_Self), "back called on empty view_interface");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+            auto _Last = _RANGES end(_Self);
+            return *--_Last;
+        }
+
+        template <random_access_range _Rng = _Derived>
+        _NODISCARD constexpr decltype(auto) operator[](const range_difference_t<_Rng> _Idx) {
+            auto& _Self = _Cast();
+#if _CONTAINER_DEBUG_LEVEL > 0
+            if constexpr (sized_range<_Derived>) {
+                _STL_VERIFY(_Idx < _RANGES size(_Self), "index out of range for view_interface");
+            }
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+            return _RANGES begin(_Self)[_Idx];
+        }
+        template <random_access_range _Rng = const _Derived>
+        _NODISCARD constexpr decltype(auto) operator[](const range_difference_t<_Rng> _Idx) const {
+            auto& _Self = _Cast();
+#if _CONTAINER_DEBUG_LEVEL > 0
+            if constexpr (sized_range<_Derived>) {
+                _STL_VERIFY(_Idx < _RANGES size(_Self), "index out of range for view_interface");
+            }
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+            return _RANGES begin(_Self)[_Idx];
+        }
+    };
+    // clang-format on
+
+    // ENUM ranges::subrange_kind
+    enum class subrange_kind : bool { unsized, sized };
+
+    // CLASS TEMPLATE ranges::subrange
+    // clang-format off
+    template <class _From, class _To>
+    concept _Convertible_to_non_slicing = convertible_to<_From, _To>
+        && !(is_pointer_v<decay_t<_From>>
+            && is_pointer_v<decay_t<_To>>
+            && _Not_same_as<remove_pointer_t<decay_t<_From>>, remove_pointer_t<decay_t<_To>>>);
+
+    template <class _Ty>
+    concept _Pair_like =
+        !is_reference_v<_Ty> && requires(_Ty __t) {
+        typename tuple_size<_Ty>::type;
+        requires derived_from<tuple_size<_Ty>, integral_constant<size_t, 2>>;
+        typename tuple_element_t<0, remove_const_t<_Ty>>;
+        typename tuple_element_t<1, remove_const_t<_Ty>>;
+        { _STD get<0>(__t) } -> convertible_to<const tuple_element_t<0, _Ty>&>;
+        { _STD get<1>(__t) } -> convertible_to<const tuple_element_t<1, _Ty>&>;
+    };
+
+    template <class _Ty, class _First, class _Second>
+    concept _Pair_like_convertible_from = !range<_Ty> && _Pair_like<_Ty>
+        && constructible_from<_Ty, _First, _Second>
+        && _Convertible_to_non_slicing<_First, tuple_element_t<0, _Ty>>
+        && convertible_to<_Second, tuple_element_t<1, _Ty>>;

```

Comment:
It's important to not do derived-to-base conversion on the iterator (because we do pointer arithmetic on that), but reasonably harmless for the sentinel. (LWG-3282)

Reply:
Aha, I was looking at the pre-Prague WP, thanks.